FirebaseStorageSwift/Sources/SwiftApiExtension.swift

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseStorage
+
+/// getResultCallback generates a closure that returns a Result type from a closure that returns an
+/// optional type and Error.
+private func getResultCallback<T>(
+  completion: @escaping (Result<T, Error>) -> Void
+) -> (_: T?, _: Error?) -> Void {
+  return { (metadata: T?, error: Error?) -> Void in
+    guard let metadata = metadata else {
+      guard let error = error else {
+        completion(.failure(NSError(domain: "FirebaseStorageSwift",
+                                    code: -1,
+                                    userInfo: ["Storage Result Generator":
+                                      "InternalError - Return type and Error code both nil"])))
+        return
+      }
+      completion(.failure(error))
+      return
+    }
+    completion(.success(metadata))
+  }
+}
+
+public extension StorageReference {
+  func downloadURL(completion: @escaping (Result<URL, Error>) -> Void) {
+    downloadURL(completion: getResultCallback(completion: completion))
+  }
+
+  func getData(maxSize: Int64, completion: @escaping (Result<Data, Error>) -> Void)
+    -> StorageDownloadTask {
+    return getData(maxSize: maxSize, completion: getResultCallback(completion: completion))
+  }
+
+  func getMetadata(completion: @escaping (Result<StorageMetadata, Error>) -> Void) {
+    getMetadata(completion: getResultCallback(completion: completion))
+  }
+
+  func list(withMaxResults maxResults: Int64,
```

Comment:
`withMaxResults` should have been `maxResults` to begin with. If we do want to introduce some inconsistency, this might be a good candidate :)

Reply:
I'll leave this to the API reviewers ...

Reply:
Leaning towards fixing this here since it matches `getData(maxSize)` and will end up breaking fewer people when we do fix it in the upstream library.

Reply:
After discussion with @ryanwilson, we decided to make the change all at once, since it would be confusing to see the different labels in Xcode completion:

![Screen Shot 2020-03-31 at 3 27 03 PM](https://user-images.githubusercontent.com/73870/78082305-7f224700-7367-11ea-99e0-4fcf39989431.png)